src/utils/weakCache.ts

```diff
@@ -38,3 +40,20 @@ export const setWeakCacheItem = <T>(
     deps = rest
   }
 }
+
+export const createMemoizeAtom = () => {
```

Comment:
Nitpick: I feel like `cacheAtom` is better naming, but that's obviously subjective.
 
```suggestion
export const createCacheAtomFn = () => {
```